File: LOO 2°Parcial/Tarea 5/Circle.java
package paqueteFiguras;

public class Circle extends Point2D {
    protected double radio;
    public Circle () {
        this.x = 0;
        this.y = 0;
        this.radio = 1;
    }
    public Circle (double x, double y, double radio) {
        super (x, y);
        this.radio = radio > 0 ? radio : 1;
    }
    @Override
    public String shapeName(){
        return "Circulo";
    }
    @Override
    public double calcularArea(){
        return (this.radio) * (this.radio) * (Math.PI);
    }
    @Override
    public double calcularPerimetro(){
        return (this.radio) * (2) * (Math.PI);
    }
    @Override
    public double calcularVolumen(){
        return 0;
    }
}

File: LOO 2°Parcial/Tarea 5/Circle.py
# -*- coding: utf-8 -*-
"""
Created on Sat Apr 10 14:41:47 2021

@author: usuario
"""


import Point2D as Punto

class Circle(Punto.Point2D):
    
    def __init__(self, x = 0, y = 0, radio = 0):
        
        Punto.Point2D.__init__(self, x, y)
        if (radio < 0):
            self.__radio = 0
        else:
            self.__radio = radio
            

    @property
    def radio(self):
        return self.__radio

    
    @radio.setter
    def radio(self, value):
        if (value < 0):
            self.__radio = 0
        else:
            self.__radio = value

    def shapename(self):
        return "Circulo"

    def calcularea(self):
        return self.__radio * self.__radio * 3.1416
    
    def calcularperimetro(self):
        return self.__radio * 2 * 3.1416
    
    def calcularvolumen(self):
        return 0